src/app/pages/todos/components/todo-card/todo-card.component.ts

```diff
@@ -1,25 +1,45 @@
-import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
-import { ITodo } from '@shared';
+import { ChangeDetectionStrategy, Component, Input, OnInit } from '@angular/core';
+import { IEditTodoForm, ITodo, Status, TodoFacadeService } from '@shared';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
 
 @Component({
   selector: 'app-todo-card',
   templateUrl: './todo-card.component.html',
   styleUrls: ['./todo-card.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class TodoCardComponent {
+export class TodoCardComponent implements OnInit {
+  form: FormGroup<IEditTodoForm>;
+
   @Input()
   todo: ITodo;
 
-  pin(id: string): void {
-    console.log(id);
+  constructor(private todoFacadeService: TodoFacadeService, private fb: FormBuilder) {}
+
+  get statusClass(): Status {
+    return Status[this.todo.status];
+  }
+
+  ngOnInit(): void {
+    this.form = this.fb.group({
+      title: [this.todo.title, Validators.required],
+      description: [this.todo.description, Validators.required],
+    });
+  }
+
+  editTodo(): void {
+    this.todoFacadeService.editTodo({ id: this.todo.id, ...this.form.value } as Partial<ITodo>);
+  }
+
+  togglePinned(): void {
+   this.todo.isPinned ? this.todoFacadeService.unpinTodo(this.todo.id) : this.todoFacadeService.pinTodo(this.todo.id);
```

Comment:
the same as I was saying :) don't use ternary operator for just calling methods/functions